gnomad/utils/annotations.py

```diff
@@ -1519,3 +1529,308 @@ def _create_group_dicts(
+    """
+    build_in = get_reference_genome(ht.locus).name
+    chr_in = ht.locus.contig
+
+    vrs_chrom_ids_expr = hl.literal(VRS_CHROM_IDS)
+    chrom_dict = vrs_chrom_ids_expr[build_in]
+    vrs_id = ht.info.vrs.VRS_Allele_IDs[1]
+    vrs_chrom_id = chrom_dict[chr_in]
+    vrs_start_value = ht.info.vrs.VRS_Starts[1]
+    vrs_end_value = ht.info.vrs.VRS_Ends[1]
+    vrs_state_sequence = ht.info.vrs.VRS_States[1]
+
+    ht_out = ht.annotate(
+        vrs=hl.struct(
+            _id=vrs_id,
+            location=hl.struct(
+                _id="",
+                type="SequenceLocation",
+                interval=hl.struct(start=vrs_start_value, end=vrs_end_value),
+                sequence_id=vrs_chrom_id,
+            ),
+            state=hl.struct(
+                type="LiteralSequenceExpression", sequence=vrs_state_sequence
+            ),
+        )
+    )
+    ht_out = ht_out.annotate(vrs_json=hl.json(ht_out.vrs))
+    return ht_out
+
+
+def add_gks_va(
+    ht: hl.Table,
+    label_name: str,
+    label_version: str,
+    coverage_ht: hl.Table = None,
+    ancestry_groups: list = None,
+    ancestry_groups_dict: dict = None,
+    by_sex: bool = False,
+) -> dict:
+    """
+    Add GKS VA annotations to a hail table.
+
+    Annotates the hail table with frequency information conforming to the GKS VA frequency schema.
+    If ancestry_groups or by_sex is provided, also include subcohort schemas for each cohort.
+    This annotation is added under the gks_va_freq_dict field of the table.
+    The focusAllele field is not populated, and must be filled in by the caller.
+
+    :param ht: Hail Table to parse for desired variant.
+    :param variant: String of variant to search for (chromosome, position, ref, and alt, separated by '-'). Example for a variant in build GRCh38: "chr5-38258681-C-T".
+    :param label_name: Label name to use within the returned dictionary. Example: "gnomAD".
+    :param label_version: String listing the version of the HT being used. Example: "3.1.2" .
+    :param coverage_ht: Hail Table containing coverage statistics, with mean depth stored in "mean" annotation. If None, omit coverage in return.
+    :param ancestry_groups: List of strings of shortened names of genetic ancestry groups to return results for. Example: ['afr','fin','nfe'] .
+    :param ancestry_groups_dict: Dict mapping shortened genetic ancestry group names to full names. Example: {'afr':'African/African American'} .
+    :param by_sex: Boolean to include breakdown of ancestry groups by inferred sex (XX and XY) as well.
+    :param vrs_only: Boolean to return only the VRS information and no general frequency information. Default is False.
+    :return: Dictionary containing VRS information (and frequency information split by ancestry groups and sex if desired) for the specified variant.
+    """
+    # Throw warnings if contradictory arguments passed.
+    if by_sex and not ancestry_groups:
```

Comment:
Since `get_gks` function was split into a `add_gks_va` and `add_gks_vrs`, I think this is not needed. The decision about what whether only vrs should be created is made in `gnomad_gks_batch`.